sklearn/sklearn-template/template/trainer/input_util.py

```diff
@@ -15,27 +15,110 @@
 
 """Data ingest operations."""
 
-# TODO(cezequiel): Remove when done using sns dataset
-import seaborn as sns
-
+import pandas as pd
 from sklearn import model_selection
+from google.cloud import storage
 
 from trainer import metadata
 
 
-def _feature_label_split(data, label_column):
-  return data.loc[:, data.columns != label_column], data[label_column]
+def _feature_label_split(data_df, label_column):
+  """
+  Split the DataFrame into two DataFrames including all the features, and label separately
 
+  Args:
+    data_df: (pandas.DataFrame) DataFrame the splitting to be performed on
+    label_column: (string) name of the label column
 
-def read_from_bigquery(table_name):
-  # TODO(cezequiel): Implement BQ reader
-  data = sns.load_dataset('titanic')
+  Returns:
+    A Tuple of (pandas.DataFrame, pandas.Series)
+  """
 
-  _ = table_name
-  label_column = metadata.LABEL
+  return data_df.loc[:, data_df.columns != label_column], data_df[label_column]
 
-  train, val = model_selection.train_test_split(data)
-  x_train, y_train = _feature_label_split(train, label_column)
-  x_val, y_val= _feature_label_split(val, label_column)
 
+def _data_train_test_split(data_df):
+  """
+  Split the DataFrame two subsets for training and testing
+
+  Args:
+    data_df: (pandas.DataFrame) DataFrame the splitting to be performed on
+
+  Returns:
+    A Tuple of (pandas.DataFrame, pandas.Series, pandas.DataFrame, pandas.Series)
+  """
+
+  label_column = metadata.LABEL
+  train, val = model_selection.train_test_split(data_df)
+  x_train, y_train = _feature_label_split(train, label_column)
+  x_val, y_val = _feature_label_split(val, label_column)
   return x_train, y_train, x_val, y_val
+
+
+def read_from_bigquery(full_table_path, project_id=None):
+  """
+  Read training data from BigQuery given full path of BigQuery table.
+
+  Args:
+    full_table_path: (string) full path of the table containing training data in the
+          format of [project_id.dataset_name.table_name]
+    project_id: (string, Optional) Google BigQuery Account project ID
+
+  Returns:
+    A Tuple of (pandas.DataFrame, pandas.Series, pandas.DataFrame, pandas.Series)
+  """
+
+  query = '''
+    Select * From `{}`
+  '''.format(full_table_path)
+
+  # Use "application default credentials"
+  # Use SQL syntax dialect
+  data_df = pd.read_gbq(query, project_id=project_id, dialect='standard')
+
+  return _data_train_test_split(data_df)
+
+
+def read_from_gcs(file_pattern, bucket_name):
```

Comment:
Consider using `tensorflow.gfile`, e.g.
```
with tf.gfile.Open('gs://bucket/path/to/file') as f:
  df = pd.read_csv(f)
```

Though I understand it must be a bit awkward to use scikit-learn and then import a TensorFlow module.
I wish they had added that module in `storage` instead.

Reply:
Please refer to my reply in PR #5

Reply:
Added comment in PR#5